packages/mui-base/src/Popover/Root/usePopoverRoot.ts

```diff
@@ -39,6 +41,7 @@ export function usePopoverRoot(params: usePopoverRoot.Parameters): usePopoverRoo
   const [descriptionId, setDescriptionId] = React.useState<string>();
   const [triggerElement, setTriggerElement] = React.useState<Element | null>(null);
   const [positionerElement, setPositionerElement] = React.useState<HTMLElement | null>(null);
+  const [openMethod, setOpenMethod] = React.useState<PointerType | null>(null);
```

Comment:
Would be great for this to be a ref instead to avoid an extra re-render, but I suppose there's no way to avoid reading it during render instead of in an effect/event handler?

Reply:
When testing it, I never saw this state causing a rerender by itself. It's always batched with other state setters.